lib/src/layer/tile_layer/tile_model.dart

```diff
@@ -0,0 +1,39 @@
+import 'dart:math';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:flutter_map/src/layer/tile_layer/tile_painter.dart';
+
+/// Model for tiles displayed by [TileLayer] and [TilePainter]
+class TileModel {
+  /// Controls how this tile is replaced by another tile
+  ///
+  /// See also:
+  ///
+  ///  * [Widget.key]
+  ///  * The discussions at [Key] and [GlobalKey].
+  final ObjectKey key;
+
+  /// [TileImage] is the model class that contains meta data for the Tile image.
+  final TileImage tileImage;
+
+  /// The [TileBuilder] is a reference to the [TileLayer]'s
+  /// [TileLayer.tileBuilder].
+  final TileBuilder? tileBuilder;
```

Comment:
`TileBuilder` is never evaluated, and has been replaced by `tilePaint`. It should be deprecated and removed from here.

Reply:
Yes